### PR TITLE
[WIP] feature/ST-524-field-is-not-disabled-on-safari

### PR DESCRIPTION
**Pull request by pgs-gmartowski**

### CODE REVIEW

File: src/ST.ts [hunk 1]
```diff
@@ -182,7 +182,7 @@ class ST {
         'font-size: 2em; font-weight: bold',
         'font-size: 2em; font-weight: 1000; color: #e71b5a',
         'font-size: 2em; font-weight: bold',
-        'font-size: 2em; font-weight: regular; color: #e71b5a',
+        'font-size: 2em; font-weight: regular; color: #e71b5a'
```
Comment (pgs-gmartowski) [inline src/ST.ts]: info: I added task for make one code style for our IDE's (to avoid such things)

File: src/core/shared/MessageBus.ts [hunk 2]
```diff
@@ -81,7 +81,10 @@ export class MessageBus {
 
   public publishFromParent(event: IMessageBusEvent, frameName: string) {
     // @ts-ignore
-    window.frames[frameName].postMessage(event, this._frameOrigin);
+    if (window.frames[frameName]) {
```
Comment (pgs-gmartowski) [inline src/core/shared/MessageBus.ts]: info: @wirus15 I've merged your changes into my branch

File: src/json/config.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "deferInit": false,
   "fieldsToSubmit": ["pan", "expirydate", "securitycode"],
   "buttonId": "merchant-submit-button",
-  "jwt": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJhbTAzMTAuYXV0b2FwaSIsImlhdCI6MTU4MTEwMTA5MC4yNDEwMTA3LCJwYXlsb2FkIjp7ImJhc2VhbW91bnQiOiIxMDAwIiwiYWNjb3VudHR5cGVkZXNjcmlwdGlvbiI6IkVDT00iLCJjdXJyZW5jeWlzbzNhIjoiR0JQIiwic2l0ZXJlZmVyZW5jZSI6InRlc3RfamFtZXMzODY0MSIsImxvY2FsZSI6ImVuX0dCIn19.EkSLEo_aEPd43AmPiYr-P0VjsGeFCS6ZT7K3_BoojIU",
+  "jwt": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJhbTAzMTAuYXV0b2FwaSIsImlhdCI6MTU4MTQyMzY2Ni44MDUwMywicGF5bG9hZCI6eyJiYXNlYW1vdW50IjoiMTAwMCIsImFjY291bnR0eXBlZGVzY3JpcHRpb24iOiJFQ09NIiwiY3VycmVuY3lpc28zYSI6IkdCUCIsInNpdGVyZWZlcmVuY2UiOiJ0ZXN0X2phbWVzMzg2NDEiLCJsb2NhbGUiOiJlbl9HQiJ9fQ.es6dGpSl6D8w8DAneHNn85vyyxSznez8XrqNkjOOPlk",
```
Comment (pgs-gmartowski) [inline src/json/config.json]: info: @wirus15 we must finally prevent this kind of files for committing

File: src/ST.ts (hunk 1 above)
Comment (wirus15) [reply src/ST.ts]: we should configure tslint/eslint to check and fix this stuff automatically

File: src/components/control-frame/ControlFrame.ts [hunk 4]
```diff
@@ -347,8 +349,13 @@ export class ControlFrame extends Frame {
   }
 
   private _getSecurityCodeLength(): number {
+    const cardDetails = JwtDecode(StCodec.jwt) as any;
```
[the remark this answers is not in the document]
Comment (pgs-gmartowski) [reply src/components/control-frame/ControlFrame.ts]: yes, it's called `IDecodedJwt`, I forgot about it

[the remark this answers is not in the document]
Comment (pgs-gmartowski) [reply src/components/control-frame/ControlFrame.ts]: done

File: src/components/control-frame/ControlFrame.ts [hunk 5]
```diff
@@ -349,7 +349,8 @@ export class ControlFrame extends Frame {
   }
 
   private _getSecurityCodeLength(): number {
-    const cardDetails = JwtDecode(StCodec.jwt) as any;
+    const cardDetails: IDecodedJwt = JwtDecode(StCodec.jwt);
+    console.error(cardDetails);
```
Comment (wirus15) [inline src/components/control-frame/ControlFrame.ts]: this probably should not be here 😁

**Comment by pgs-gmartowski** [comment]
> I'm just wondering if anything could be done about this `any`?

Yes, I changed this to `IDecodedJwt`

File: src/components/control-frame/ControlFrame.ts (hunk 5 above)
Comment (pgs-gmartowski) [reply src/components/control-frame/ControlFrame.ts]: oops :) deleted

**Pull request merged**